ci/build_cpp.sh

```diff
@@ -15,7 +15,11 @@ rapids-print-env
 
 rapids-logger "Begin cpp build"
 
+sccache --zero-stats
+
 RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rapids-conda-retry mambabuild \
     conda/recipes/libcuspatial
 
+sccache --show-adv-stats
```

Comment:
Added `sccache` stats printing for conda builds as well based on https://github.com/rapidsai/build-planning/issues/111 and offline conversation with @vyasr .

### Why not `build.sh`?

Here, I'm proposing that these be put in `ci/build_{cpp,python}.sh`, wrapped around any `conda mambabuild` invocations that will end up compiling anything (so, for example, I included them in `build_python.sh` because `cuspatial` and `cuproj` both have some Cython extensions).

I think this is preferable to `build.sh`, because the `build_{cpp,python}.sh` scripts are unambiguously for CI, and can unconditionally invoke `sccache`. `build.sh` is invoked in CI, but it's also intended for local use, and I didn't want to assume the presence of `sccache` for all local builds are introduce the complexity of having to check for its use.

There are also some projects, like `cuvs`, that have their own handling of printing `sccache` stats and creating build summary reports based on them in `build.sh`, and I didn't want to interfere with that.

example:

https://github.com/rapidsai/cuvs/blob/e7f1085b71c340b9600f5f38f7f0059a5c7aa806/build.sh#L364-L366

https://github.com/rapidsai/cuvs/blob/e7f1085b71c340b9600f5f38f7f0059a5c7aa806/build.sh#L408-L424